iocore/net/SSLSessionTicket.h

```diff
@@ -0,0 +1,42 @@
+/** @file
+
+  SessionTicket TLS Extension
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#pragma once
+
+#include <openssl/safestack.h>
+#include <openssl/tls1.h>
+
+// Check if the ticket_key callback #define is available, and if so, enable session tickets.
+#ifdef SSL_CTX_set_tlsext_ticket_key_cb
+#define HAVE_OPENSSL_SESSION_TICKETS 1
+#endif
+
+#ifdef HAVE_OPENSSL_SESSION_TICKETS
+
+#include <openssl/crypto.h>
+#include <openssl/hmac.h>
```

Comment:
I removed checking `HAVE_OPENSSL_HMAC_H`.

1.) Defining `ssl_callback_session_ticket` without `<openssl/hmac>` doesn't make scene.
2.) There no macro/script to define `HAVE_OPENSSL_HMAC_H`. There should be `AC_CHECK_HEADERS(openssl/hmac.h)` or something similar, but I couldn't find such code. All I found is a check for hmac, but I think this doesn't define the macro.

https://github.com/apache/trafficserver/blob/426640346798589dcd7d5b87c9f64e748208ceaf/configure.ac#L1320

Prior this change,  checking `HAVE_OPENSSL_HMAC_H` have never worked and `<openssl/hmac.h>` was include from somewhere in dependency.